src/main/kotlin/uk/gov/gdx/datashare/service/LegacyAdaptorInbound.kt

```diff
@@ -41,71 +42,80 @@ class LegacyAdaptorInbound(
     try {
       runBlocking {
         LockAssert.assertLocked()
-        S3Client {
-          region = "eu-west-2"
-        }.use { s3 ->
-          log.debug("Polling S3 bucket: $ingressBucket")
-          val listObjectsRequest = ListObjectsRequest {
-            bucket = ingressBucket
-          }
-          val objects = s3.listObjects(listObjectsRequest).contents
-
-          objects?.forEach { s3Object ->
+        log.debug("Polling S3 bucket: ${s3Config.ingressBucket}")
+        amazonS3.listObjectsV2(s3Config.ingressBucket).objectSummaries
+          .forEach { s3Object ->
             s3Object.key?.let {
-              processBucketObject(s3, it)
+              processBucketObject(it)
             }
           }
-        }
       }
-    } catch (e: NoSuchMethodError) {
-      log.error("Failed to connect to S3")
-    } catch (e: Exception) {
+    } catch (e: AmazonClientException) {
       log.error("Failed to connect to S3", e)
     }
   }
 
-  suspend fun postDataToReceiver(payload: EventToPublish): ResponseEntity<Void> =
-    dataReceiverApiWebClient.post()
-      .uri("/event-data-receiver")
-      .bodyValue(payload)
-      .retrieve()
-      .awaitBodilessEntity()
+  suspend fun processBucketObject(objectKey: String) {
 
-  // TODO-https://github.com/alphagov/gdx-data-share-poc/issues/97: Fix S3 client
-  suspend fun processBucketObject(s3: S3Client, objectKey: String) {
-    val getObjectRequest = GetObjectRequest {
-      key = objectKey
-      bucket = ingressBucket
-    }
-    val encodedCopySource = withContext(Dispatchers.IO) {
-      URLEncoder.encode("$ingressBucket/$objectKey", StandardCharsets.UTF_8.toString())
-    }
-    val copyObjectRequest = CopyObjectRequest {
-      copySource = encodedCopySource
-      key = objectKey
-      bucket = archiveIngressBucket
-    }
-    val deleteObjectRequest = DeleteObjectRequest {
-      key = objectKey
-      bucket = ingressBucket
+    try {
+      log.debug("Retrieving object from S3: $objectKey")
+      amazonS3.getObject(s3Config.ingressBucket, objectKey)
+        .also {
+          log.info("Found file for processing ${it.key}")
+        }.objectContent
+        .let {
+          getCSV(it).forEach { csvRecord ->
+            with(csvRecord) {
+              postDataToReceiver(
+                EventToPublish(
+                  id = certificateId(),
+                  eventType = "DEATH_NOTIFICATION",
+                  eventDetails = "${lastName()},${firstName()},${dateOfBirth()},${dateOfDeath()},${gender()},\"${address()}\"" // TODO: decide whether to format etc
```

Comment:
I think I'm _expecting_ this to be JSON-like, but I don't think we do much with it so it probably doesn't matter right now